src/module/system/damage/roll-dialog.ts

```diff
@@ -159,44 +158,29 @@ export class DamageRollModifiersDialog extends Application {
             diceResults: {},
             baseDamageDice: damage.effectDice,
         };
-        const renderData: DamageRollRenderData = {
-            formula: formula.formula,
-            damageTypes: {},
-            total: 1,
-        };
+
         const rolls: Rolled<Roll>[] = [];
 
         for (const [damageType, categories] of Object.entries(formula.partials)) {
             rollData.diceResults[damageType] = {};
-            renderData.damageTypes[damageType] = {
-                categories: {},
-                icon: DAMAGE_TYPE_ICONS[damageType] ?? damageType,
-                label: damageTypes[damageType as DamageType] ?? damageType,
-            };
             for (const [damageCategory, partial] of Object.entries(categories)) {
-                const data: object = formula.data;
-                const roll = await new Roll(partial, data).evaluate({ async: true });
+                const roll = await new Roll(partial, formula.data).evaluate({ async: true });
                 rolls.push(roll);
                 const damageValue = rollData.types[damageType] ?? {};
                 damageValue[damageCategory] = roll.total;
                 rollData.types[damageType] = damageValue;
                 rollData.total += roll.total;
-                rollData.diceResults[damageType][damageCategory] = [];
-                renderData.damageTypes[damageType].categories[damageCategory] = {
-                    dice: roll.dice.flatMap((d) =>
-                        d.results.map((r) => {
-                            // Record the partial result for the roll flag
-                            rollData.diceResults[damageType][damageCategory].push(r.result);
-                            // Return the needed render data
-                            return { faces: d.faces ?? 0, result: r.result };
-                        })
-                    ),
-                    formula: partial,
-                    label: damageCategories[damageCategory as DamageCategory] ?? damageCategory,
-                    total: roll.total,
-                };
+                rollData.diceResults[damageType][damageCategory] = roll.dice.flatMap((d) =>
+                    d.results.map((r) => ({ faces: d.faces ?? 0, result: r.result }))
+                );
```

Comment:
This is currently a `number` array of roll results that is stored in the message flag. I wasn't sure if that is expected to be there by some modules so I left it in when I changed the render method a few weeks ago.
I guess changing it is the fastest way to find out. 😁

Reply:
It is currently that yeah. Unfortunately the render data and the result data are currently created simultaneously, and if we want a better separation of concerns, we need to be able to create the render data from the result data and with the current format that's not very possible. I'm just hoping it doesn't break.....too much....